src/test/java/com/rtb/projectmanagementtool/project/ProjectControllerTest.java

```diff
@@ -18,18 +27,110 @@ public void tearDown() {
+    int projectsSize = projects.size();
+    Assert.assertTrue(projectsSize == 2);
+
+    for (ProjectData project : projects) {
+      projectController.removeProject(project);
+      Assert.assertTrue(
+          projectController.getProjects(Collections.emptyList()).size() == projectsSize - 1);
+      projectsSize--;
+    }
+  }
+
+  @Test
+  public void getAllProjects() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ProjectController projectController = new ProjectController(datastore);
+
+    HashSet<ProjectData> expectedProjects = new HashSet<ProjectData>();
 
-  @Ignore
-  public void getProjectsWithUser() {}
+    ProjectData project1 = new ProjectData("Project1", "Project1 Desc", 1l);
+    projectController.saveProject(project1);
+    expectedProjects.add(project1);
 
-  @Ignore
-  public void addProjects() {}
+    ProjectData project2 = new ProjectData("Project2", "Project2 Desc", 1l);
+    projectController.saveProject(project2);
+    expectedProjects.add(project2);
 
-  @Ignore
-  public void removeProject() {}
+    ProjectData project3 = new ProjectData("Project3", "Project3 Desc", 2l);
+    projectController.saveProject(project3);
+    expectedProjects.add(project3);
 
-  @Ignore
-  public void saveProjects() {}
+    ProjectData project4 = new ProjectData("Project4", "Project4 Desc", 3l);
+    projectController.saveProject(project4);
+    expectedProjects.add(project4);
+
+    HashSet<ProjectData> actualProjects = projectController.getProjects(Collections.emptyList());
+
+    // Can't do assertEquals on the actual arrays because of ordering in respective sets
+    Assert.assertEquals(actualProjects.size(), expectedProjects.size());
+  }
+
+  // get projects that a particular user is in
+  @Test
+  public void getProjectsWithUser() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ProjectController projectController = new ProjectController(datastore);
+
+    final Long desiredUser = 5l;
+
+    HashSet<ProjectData> expectedProjects = new HashSet<ProjectData>();
+
+    ProjectData project1 = new ProjectData("Project1", "Project1 Desc", 1l);
+    project1.addRegularUser(desiredUser);
+    projectController.saveProject(project1);
+    expectedProjects.add(project1);
+
+    ProjectData project2 = new ProjectData("Project2", "Project2 Desc", 1l);
+    project2.addAdminUser(desiredUser);
+    projectController.saveProject(project2);
+    expectedProjects.add(project2);
+
+    ProjectData project3 = new ProjectData("Project3", "Project3 Desc", 2l);
+    projectController.saveProject(project3);
+
+    ProjectData project4 = new ProjectData("Project4", "Project4 Desc", 3l);
+    projectController.saveProject(project4);
+
+    HashSet<ProjectData> actualProjects =
+        projectController.getProjects(new ArrayList<Long>(Arrays.asList(desiredUser)));
+
+    Assert.assertEquals(actualProjects.size(), expectedProjects.size());
```

Comment:
Can you also assert that the correct projects were returned and not just the correct amount?